### PR TITLE
Feature/group api

### LINKED ISSUE #63: Schnittstelle um Root-Dir URL zu bekommen
`GET /material1/group/{group_id}/url`
Gibt als JSON/Plaintext die URL des Root-Dirs der Gruppe zurück:
`/material1/dir/{dir_id}`

### PR DESCRIPTION
**Pull request by Segelzwerg**
Adds requested external ID.

closes #63

### CODE REVIEW

File: src/main/java/mops/businesslogic/GroupURLWrapper.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
+package mops.businesslogic;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+@SuppressWarnings("PMD")
```
Comment (iTitus) [inline src/main/java/mops/businesslogic/GroupURLWrapper.java]: what PMD errors are you suppressing here?

File: src/main/java/mops/businesslogic/GroupURLWrapper.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+     */
+    @JsonProperty("group_id")
+    private final long groupId;
+    /**
+     * The relative url of the group root dir.
+     */
+    @JsonProperty("url")
+    private final String url;
+
+
```
Comment (iTitus) [inline src/main/java/mops/businesslogic/GroupURLWrapper.java]:
```suggestion

```

File: src/test/java/mops/presentation/GroupControllerTest.java [hunk 3]
```diff
@@ -65,13 +64,26 @@
...
 
+    /**
+     * Tests the API for getting the group url.
+     */
+    @Test
+    public void getGroupURL() throws Exception {
+        setupSecurityContextMock(account);
+        mvc.perform(get("/material1/group/1/url"))
+                .andExpect(status().is2xxSuccessful())
+                //.andReturn();
```
Comment (iTitus) [inline src/test/java/mops/presentation/GroupControllerTest.java]:
```suggestion
```

File: src/test/java/mops/presentation/GroupControllerTest.java [hunk 4]
```diff
@@ -65,13 +64,26 @@
...
+                .andExpect(jsonPath("$.url").value("/dir/1"));
```
Comment (iTitus) [inline src/test/java/mops/presentation/GroupControllerTest.java]: misses test for second value group_id

File: src/main/java/mops/businesslogic/GroupURLWrapper.java [hunk 5]
```diff
@@ -0,0 +1,25 @@
+package mops.businesslogic;
+
+import com.fasterxml.jackson.annotation.JsonGetter;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import lombok.AllArgsConstructor;
+
+/**
+ * Ignore unused private field warning
+ */
+@AllArgsConstructor
```
Comment (iTitus) [inline src/main/java/mops/businesslogic/GroupURLWrapper.java]: can be @Value

File: src/main/java/mops/businesslogic/GroupURLWrapper.java [hunk 6]
```diff
@@ -0,0 +1,25 @@
+package mops.businesslogic;
+
+import com.fasterxml.jackson.annotation.JsonGetter;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import lombok.AllArgsConstructor;
+
+/**
+ * Ignore unused private field warning
+ */
```
Comment (iTitus) [inline src/main/java/mops/businesslogic/GroupURLWrapper.java]: should be a real javadoc or just a comment

File: src/main/java/mops/businesslogic/GroupURLWrapper.java [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+@AllArgsConstructor
+public class GroupURLWrapper {
+    /**
+     * The id of the group.
+     */
+    @JsonProperty("group_id")
+    private final long groupId;
+
+    /**
+     * The relative url of the group root dir.
```
Comment (iTitus) [inline src/main/java/mops/businesslogic/GroupURLWrapper.java]: checkstyle is complaining about ill-formatted javadoc

File: src/test/java/mops/presentation/GroupControllerTest.java [hunk 8]
```diff
@@ -65,13 +64,26 @@
...
+    /**
+     * Tests the API for getting the group url.
+     */
+    @Test
+    public void getGroupURL() throws Exception {
+        setupSecurityContextMock(account);
+        mvc.perform(get("/material1/group/1/url"))
+                .andExpect(status().is2xxSuccessful())
+                .andExpect(jsonPath("$.url").value("/dir/1"))
+                .andExpect(jsonPath("$.group_id").value("1"));
```
Comment (iTitus) [inline src/test/java/mops/presentation/GroupControllerTest.java]:
```suggestion
                .andExpect(jsonPath("$.group_id").value(1));
```

File: src/main/java/mops/businesslogic/GroupURLWrapper.java [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+
+import com.fasterxml.jackson.annotation.JsonGetter;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import lombok.AllArgsConstructor;
+
+/**
+ * Wraps a url for a group given a group id.
+ */
+@AllArgsConstructor
+public class GroupURLWrapper {
```
Comment (iTitus) [inline src/main/java/mops/businesslogic/GroupURLWrapper.java]: suggested change: `GroupDirUrlWrapper`

File: src/test/java/mops/presentation/GroupControllerTest.java (hunk 8 above)
Comment (Segelzwerg) [reply src/test/java/mops/presentation/GroupControllerTest.java]:
```suggestion
                .andExpect(jsonPath("$.group_id").value(1L));
```
we could actually use `1L`

File: doc/documentation/03_system_scope_and_context.adoc [hunk 10]
```diff
@@ -9,9 +9,17 @@ _Festlegung *aller* Kommunikationsbeziehungen mit Erklärung der *fachlichen* Ei
...
 |Kommunikationsbeziehung |Eingabe | Ausgabe
 | Requests vom Gruppensystem | _Wir erhalten Requests für die Ordner-/Dateierstellung oder Löschung und den Erhalt von Dateien aus unserer Datenbank._ | _Wir liefern eine Bestätigung des betreffenden Requests (eventuell inklusive der gewollten Datei) zurück, wenn die entsprechenden Berechtigungen vorhanden waren. Ansonsten bearbeiten wir den Request nicht._
 | Erfragen von Berechtigungen | _Wir erfragen mit einer Gruppen ID und dem Benutzernamen die Berechtigungen eines Nutzers in einer bestimmten Gruppe._  | _Wir erhalten ein Datenobjekt (.json) zurück, dass alle Berechtigungen (Rollen) des betreffenden Nutzers beinhaltet._
 |===
+
+****
+[cols="1,1,1" options="header"]
+|===
+|Anfrage |Route |Erläuterung
+|Gruppenurl |`/material1/group/{groupId}/url | Gibt eine JSON Objekt zurück, welches die Attribute `group_id` und `url` hat. Zweiteres ist der relative Pfad nach `material1/`, wo der Rootordner der Gruppe aufgerufen werden kann.
```
Comment (iTitus) [inline doc/documentation/03_system_scope_and_context.adoc]:
```suggestion
|Gruppenurl |`/material1/group/{groupId}/url | Gibt ein JSON-Objekt zurück, das die Attribute `group_id` und `url` hat. Letzteres ist der relative Pfad, wo der Rootordner der Gruppe aufgerufen werden kann.
```

File: src/main/java/mops/businesslogic/GroupDirUrlWrapper.java [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+     */
+    @JsonProperty("group_id")
+    private final long groupId;
+
+    /**
+     * @return the relative url of the group root dir.
+     */
+    @JsonGetter("url")
+    public String getGroupUrl() {
+        return String.format("/dir/%d", groupId);
```
Comment (iTitus) [inline src/main/java/mops/businesslogic/GroupDirUrlWrapper.java]:
```suggestion
        return String.format("/material1/dir/%d", groupId);
```

File: src/main/java/mops/businesslogic/GroupService.java [hunk 12]
```diff
@@ -14,4 +14,11 @@
      * @return a list of group ids
      */
     List<Directory> getAllGroupRootDirectories(Account account);
+
+    /**
+     * @param account user account
+     * @param groupId the id of the group
+     * @return a wrapper for group urls
+     */
+    GroupDirUrlWrapper getGroupURL(Account account, long groupId);
```
Comment (iTitus) [inline src/main/java/mops/businesslogic/GroupService.java]:
```suggestion
    GroupDirUrlWrapper getGroupUrl(Account account, long groupId);
```

File: src/main/java/mops/presentation/GroupController.java [hunk 13]
```diff
@@ -39,6 +43,21 @@ public String getAllFilesOfDirectory(KeycloakAuthenticationToken token,
...
 
+    /**
+     * @param token   a keycloak authentication token
+     * @param model   spring boot view model
+     * @param groupId the id of the group of the requested url
+     * @return a wrapper for the url string
+     */
+    @GetMapping(value = "/{groupId}/url", produces = MediaType.APPLICATION_JSON_VALUE)
+    @ResponseBody
+    public GroupDirUrlWrapper getGroupURL(KeycloakAuthenticationToken token,
```
Comment (iTitus) [inline src/main/java/mops/presentation/GroupController.java]:
```suggestion
    public GroupDirUrlWrapper getGroupUrl(KeycloakAuthenticationToken token,
```

File: src/main/java/mops/presentation/GroupController.java [hunk 14]
```diff
@@ -39,6 +43,21 @@ public String getAllFilesOfDirectory(KeycloakAuthenticationToken token,
...
+                                          Model model,
+                                          @PathVariable("groupId") long groupId) {
+        Account account = AccountUtil.getAccountFromToken(token);
+        return groupService.getGroupURL(account, groupId);
```
Comment (iTitus) [inline src/main/java/mops/presentation/GroupController.java]:
```suggestion
        return groupService.getGroupUrl(account, groupId);
```

File: src/test/java/mops/presentation/GroupControllerTest.java [hunk 15]
```diff
@@ -65,13 +64,26 @@
     void setUp() {
         account = new Account("studi", "bla@bla.de", "studentin");
         given(fileService.getAllFilesOfGroup(account, 1)).willReturn(List.of());
+        given(groupService.getGroupURL(account, 1)).willReturn(new GroupDirUrlWrapper(1L));
```
Comment (iTitus) [inline src/test/java/mops/presentation/GroupControllerTest.java]:
```suggestion
        given(groupService.getGroupUrl(account, 1)).willReturn(new GroupDirUrlWrapper(1L));
```

File: src/test/java/mops/presentation/GroupControllerTest.java [hunk 16]
```diff
@@ -65,13 +64,26 @@
...
                 .alwaysDo(print())
                 .apply(springSecurity())
                 .build();
     }
 
+    /**
+     * Tests the API for getting the group url.
+     */
+    @Test
+    public void getGroupURL() throws Exception {
```
Comment (iTitus) [inline src/test/java/mops/presentation/GroupControllerTest.java]:
```suggestion
    public void getGroupUrl() throws Exception {
```

**Pull request merged**